src/components/Asset/Edit/EditMetadata.tsx

```diff
@@ -78,17 +80,43 @@ export default function Edit({
         values.price !== asset.accessDetails.price &&
         (await updateFixedPrice(values.price))
 
+      if (values.files[0]?.url) {
+        const file = {
+          nftAddress: asset.nftAddress,
+          datatokenAddress: asset.services[0].datatokenAddress,
+          files: [
+            {
+              type: 'url',
+              index: 0,
+              url: values.files[0].url,
+              method: 'GET'
+            }
+          ]
+        }
+        const filesEncrypted = await getEncryptedFiles(
+          file,
+          asset.services[0].serviceEndpoint
+        )
+        updatedFiles = filesEncrypted
+      }
       const updatedService: Service = {
         ...asset.services[0],
-        timeout: mapTimeoutStringToSeconds(values.timeout)
+        timeout: mapTimeoutStringToSeconds(values.timeout),
+        files: updatedFiles
       }
 
+      // TODO: remove version update at a later time
       const updatedAsset: Asset = {
```

Comment:
on first look seems we could have spotted this issue earlier if we would use the `DDO` typing here instead?

```ts
const updatedAsset: DDO
```

Reply:
nope, typescript does not remove the props. `Asset`  doesn't have accessDetails and yet it's still there.